src/averageAges.js

```diff
@@ -55,7 +67,26 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothersWithChild = people.filter(({ mother, sex }) =>
+    onlyWithSon
+      ? hasMother(mother, people) && sex === 'm'
+      : hasMother(mother, people)
```

Comment:
you can move this logic inside of the hasMother function. make it accept third parameter and set it to false by default